pyrdp/core/ssl.py

```diff
@@ -27,9 +27,16 @@ class ClientTLSContext(ssl.ClientContextFactory):
     """
 
     def getContext(self):
-        context = SSL.Context(SSL.TLSv1_METHOD)
+        # Allow the MITM to connect to an RDP Server with ANY TLS version supported by the installed
+        # OpenSSL version. See https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=784153
+        # It was removed from OpenSSL, but PyOpenSSL has not changed their constant names yet.
+        context = SSL.Context(SSL.SSLv23_METHOD)
         context.set_options(SSL.OP_DONT_INSERT_EMPTY_FRAGMENTS)
         context.set_options(SSL.OP_TLS_BLOCK_PADDING_BUG)
+
+        # We disable TLS 1.3 because the way to decrypt TLS 1.3 traffic differs from
+        # previous TLS versions and is not yet supported by PyRDP.
```

Comment:
is it really pyrdp that doesn't support 1.3, I thought it was more OpenSSL's master secret extraction that was lagging behind?

Reply:
OpenSSL (starting from v1.1.1) supports TLS 1.3 and the necessary methods to extract the needed secrets/keys (but its different from <TLS1.3). PyOpenSSL doesnt seem to expose them, but I could be wrong. For this reason,  I prefer to say that PyRDP doesnt support TLS 1.3, because I'm not sure if PyOpenSSL is really lagging behind.

Supporting TLS 1.3 would also probably increase the required OpenSSL version to 1.1.1, and i'm not sure of the different package manager's versions of OSSL. More investigation is required

Reply:
Fair enough.